lib/AST/DeclBase.cpp

```diff
@@ -1371,7 +1375,7 @@ static bool shouldBeHidden(NamedDecl *D) {
   // from being visible?
   if (isa<ClassTemplateSpecializationDecl>(D))
     return true;
-  if (FunctionDecl *FD = dyn_cast<FunctionDecl>(D))
+  if (auto *FD = dyn_cast<FunctionDecl>(D))
```

Comment:
Is this change intentional?

Reply:
Yes, it is. We should avoid repetition of the type which is already there in the `dyn_cast`'s template params.

Reply:
The upstream has `auto *` as well.